fnl/conjure/cljdocs/db.fnl

```diff
@@ -0,0 +1,150 @@
+            fennel conjure.aniseed.fennel
+            util conjure.cljdocs.util
+            raw conjure.cljdocs.raw
+            parse conjure.cljdocs.parse
+            sql sql}})
+
+;; Schemas -------------------------------------------------------------------------------------
+;; ---------------------------------------------------------------------------------------
+(def- clj-schema
+  (let [keys ["id integer primary key"
+              "ns text"
+              "name text"
+              "symbol text"
+              "arglists text"
+              "doc text"
+              "preview text"
+              "examples text"
+              "macro int"
+              "see_alsos text"
+              "static int"
+              "type text"
+              "notes text"]]
+    (string.format "create table if not exists clj(%s)"
+                   (str.join ", " keys))))
+
+(defn- get-schema [ext]
+  (match ext
+    "clj" clj-schema))
+
+;; db -------------------------------------------------------------------------------------
+;; ---------------------------------------------------------------------------------------
+(defn- init-db [db]
+  (let [keys ["id integer"
+              "extension text not null"
+              "init int not null"]]
+    (db:eval (string.format "create table if not exists state(%s)"
+                            (str.join ", " keys)))
+    (db:insert "state"
+              [{:id 1
+                :extension "clj"
+                :init 0}
+               {:id 2
+                :extension "cljs"
+                :init 0}])))
+
+(def- dbpath
+ (.. (util.cache-dir)
+     "/usage_docs.db"))
+
+(defn- db-exists? []
+  (util.exists? dbpath))
+
+(def- db
+  (let [db (sql.new dbpath)
+        init (if (not (db-exists?)) init-db (fn [db]))]
+    (db:with_open init)))
+
+
+;; format --------------------------------------------------------------------------------
+;; ---------------------------------------------------------------------------------------
+(defn- list->string [v]
+  (if (= "table" (type v))
+    (str.join "\n|00|\n " v)
+    v))
+
+
+(defn- boolean->int [v]
+  (if (= "boolean" (type v))
+    (if (= v true) 1 0)
+    v))
+
+(defn- vimnil->nil [v]
+  (if (= "userdata" (type v))
+    nil
+    v))
+
+(defn- fix-datatypes [v]
+  (-> v
+      list->string
+      boolean->int
+      vimnil->nil))
+
+(defn- format-clj-entry [item]
+  (let [markdown {:preview (parse.parse-for "clj" item)}
+        res {}]
+    (a.merge! item markdown)
+    (each [k v (pairs item)]
+      (tset res (k:gsub "-" "_") (fix-datatypes v)))
+    res))
+
+;; seed ----------------------------------------------------------------------------------
+;; ---------------------------------------------------------------------------------------
+
+(defn- seed-clj [cb]
+  (cb (db:with_open
+        #(let [items (a.map format-clj-entry (raw.get "clj"))]
+           (db:eval (get-schema "clj"))
+           (each [_ item (ipairs items)] ;; would work without looping
```

Comment:
Results in `runtime error: Wrong number of arguments.` error with only inserting 17 items